fsm.go

```diff
@@ -120,15 +119,8 @@ func (s *stateMachine) timeout() []packet {
 		return nil
 	}
 	var ps []packet
-	for _, i := range rand.Perm(len(s.ml.order)) {
-		id := s.ml.order[i]
-		if id == s.pingTarget {
-			continue
-		}
+	for _, id := range s.ml.o.Rand(s.nPingReqs, s.pingTarget) {
```

Comment:
I feel like `o` isn't a great name for that field. (And same goes for `ml` for that matter.)

Like what is `s.ml.o`? I think seeing something like `s.members.order` here would be way more helpful.

My rule of thumb is that single-letter variable names are ok for loop indexes and local variables of very short functions or when the variable is used on almost every line. For identifiers that have a larger scope and are used less frequently, the name needs to convey more information.

From https://research.swtch.com/names:

> Global names must convey relatively more information, because they appear in a larger variety of contexts.

Reply:
The memberList type has been asking to be refactored for a while now, and its fields have been the subjects of a few naming decisions that haven't stood the test of time. For example, `s.members` would look good except that type memberList has a field called `members` (which is a map of messages - a glaring design holdover from the initial sketch of the package). There's a lot of tidying up to do, enough that I think it should be the subject of its own PR.